client/my-sites/media/main.jsx

```diff
@@ -281,26 +301,68 @@ class Media extends Component {
 		return '/media/:site';
 	};
 
+	getSelectedItems = () => {
+		if ( this.props.media ) {
+			return [ this.props.media ];
+		}
+		return this.state.selectedItems;
+	};
+	getSelectedItem = () => {
+		if ( this.props.media ) {
+			return this.props.media;
+		}
+		return this.state.selectedItems[ this.state.editedImageItem ];
+	};
+
+	getSelectedIndex = () => {
+		if ( this.props.media ) {
+			return 0;
+		}
+		return this.state.currentDetail;
+	};
+
+	showDialog = ( typeOfDialog = null ) => {
+		if ( typeOfDialog === 'detail' ) {
+			if (
+				this.props.media &&
+				this.state.editedImageItem === null &&
+				this.state.editedVideoItem === null
+			) {
+				return true;
+			}
+			return this.state.currentDetail !== null;
+		}
+
+		if ( this.props.media ) {
+			return true;
+		}
+		return (
+			this.state.editedImageItem !== null ||
+			this.state.editedVideoItem !== null ||
+			this.state.currentDetail !== null
+		);
+	};
+
 	render() {
 		const site = this.props.selectedSite;
+		const mediaId = this.props.mediaId;
 		return (
 			<div ref="container" className="main main-column media" role="main">
+				{ mediaId && site && site.ID && <QueryMedia siteId={ site.ID } mediaId={ mediaId } /> }
```

Comment:
That is true. I was hoping to only pass the slug and the site_id to the component instead of the whole site object. This would make things a lot simpler but I didn't want to make this PR even larger.